File: src/lambda/dms-switch/utils/wait-for-dms-config-status.ts
import { DatabaseMigrationServiceClient } from '@aws-sdk/client-database-migration-service';
import { getDmsReplicationStatus } from './get-dms-replication-status';

export const waitForDmsStatus = async ({
  dms,
  ReplicationConfigArn: ReplicationConfigArn,
  targetStatus,
}: {
  dms: DatabaseMigrationServiceClient;
  ReplicationConfigArn: string;
  targetStatus: string;
}): Promise<string> => {
  let status = '';
  for (let j = 0; j < 24; j++) {
    status = await getDmsReplicationStatus({ dms, ReplicationConfigArn });
    console.log(`DMS status: ${status}`);
    if (status === targetStatus) {
      return status;
    }
    await new Promise(f => setTimeout(f, 10000));
  }
  throw new Error(`DMS not ${targetStatus}: ${status}`);
};

File: src/lambda/dms-switch-py/utils/wait_for_dms_config_status.py
import time
from utils.get_dms_replication_status import get_dms_replication_status


def wait_for_dms_status(dms, replication_config_arn, target_status):
    status = ''
    for _ in range(24):
        status = get_dms_replication_status(dms, replication_config_arn)
        print(f'DMS status: {status}')
        if status == target_status:
            return status
        time.sleep(10)
    raise Exception(f'DMS not {target_status}: {status}')
